src/CommunityToolkit.Maui/Essentials/FileFolderDialog.tizen.cs

```diff
@@ -44,28 +33,36 @@ sealed class FileFolderDialog : Popup<string>, IDisposable
 	string selectedPath;
 	string selectedFileName;
 
-	public FileFolderDialog(FileSelectionMode mode, string initialPath, CancellationToken cancellationToken = default, string fileName = "default")
+	public static string TryGetExternalDirectory()
 	{
-		if (!Enum.IsDefined(typeof(FileSelectionMode), mode))
+		string? externalDirectory = null;
+		foreach (var storage in Tizen.System.StorageManager.Storages)
 		{
-			throw new InvalidEnumArgumentException($"{mode} is not valid for {nameof(FileSelectionMode)}.");
+			if (storage.StorageType == Tizen.System.StorageArea.External)
+			{
+				externalDirectory = storage.RootDirectory;
+				break;
+			}
+			else if (storage.StorageType == Tizen.System.StorageArea.Internal)
+			{
+				externalDirectory = storage.RootDirectory;
+			}
 		}
+
+		return externalDirectory!;
```

Comment:
we don't use the `!` operator. here you can do something like
```suggestion
		return externalDirectory ?? throw new InvalidOperationException("Huston we have a problem!");
```